bots/learn-tests

```diff
@@ -98,25 +98,19 @@ def train(filename, url, verbose=False):
     if verbose:
         sys.stderr.write(" ^ {0}\n".format(url))
 
-    cmd = [ "curl", "--fail", "--upload-file", filename ]
-    if verbose:
-        cmd.append("--progress-bar")
-    cmd.append(url)
+    cmd = [ os.path.join(BOTS, "image-upload"), "--state", filename, "--store", url ]
 
     # Passing through a non terminal stdout is necessary to make progress work
-    curl = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-    cat = subprocess.Popen(["cat"], stdin=curl.stdout)
-    curl.stdout.close()
-    ret = curl.wait()
-    cat.wait()
-    if ret != 0:
+    try:
+        subprocess.check_call(cmd)
+    except:
```

Comment:
No bare excepts please. `subprocess.CalledProcessError` seems fine. But why bother with that and re-raising a different exception? IMHO you can just let it fail, the original command line already has the url.

Reply:
woops, sorry about that. changed :)